### PR TITLE
Migrated some Legacy Tests to Unit Tests

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrated some Legacy Tests to Unit Tests
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #23462
| How to test?      | N/A
| Possible impacts? | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24121)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

File: tests/Integration/Classes/Assets/JavascriptManagerTest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+class JavascriptManagerTest extends TestCase
+{
+    /**
+     * @dataProvider providerIsAssets
+     */
+    public function testIsAssets(string $id, bool $toBeFound, $expectedPath): void
+    {
+        $expectedAsset = false;
+        $testsPath = '/tests-legacy/resources/assets/';
```
Comment (atomiix) [inline tests/Integration/Classes/Assets/JavascriptManagerTest.php]: Shouldn't it be moved to the `tests` folder?

File: tests/Integration/Classes/Assets/StylesheetManagerTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use StylesheetManager;
+
+class StylesheetManagerTest extends TestCase
+{
+    /**
+     * @dataProvider providerIsAssets
+     */
+    public function testIsAssets(string $id, bool $toBeFound, $expectedPath): void
+    {
+        $testsPath = '/tests-legacy/resources/assets/';
```
Comment (atomiix) [inline tests/Integration/Classes/Assets/StylesheetManagerTest.php]: Same here

File: tests/Integration/Adapter/Cart/CartPresenterTest.php [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+     *
+     * @param object &$object Instantiated object that we will run method on
+     * @param string $methodName Method name to call
+     * @param array $parameters array of parameters to pass into method
+     *
+     * @return mixed method return
+     *
+     * @see https://jtreminio.com/2013/03/unit-testing-tutorial-part-3-testing-protected-private-methods-coverage-reports-and-crap/
+     */
+    protected function invokeMethod(&$object, string $methodName, array $parameters = [])
```
Comment (sowbiba) [inline tests/Integration/Adapter/Cart/CartPresenterTest.php:170]: I see this method at least twice. Cant we have an intermediary class to implement this ?

File: tests-legacy/Unit/Core/Product/PriceFormatter.php [hunk 4]
```diff
@@ -1,42 +0,0 @@
...
- * @author    PrestaShop SA and Contributors <contact@prestashop.com>
- * @copyright Since 2007 PrestaShop SA and Contributors
- * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
- */
-
-namespace LegacyTests\Unit\Core\Product;
-
-use PrestaShop\PrestaShop\Adapter\Product\PriceFormatter as BasePricePresenter;
-
-class PriceFormatter extends BasePricePresenter
```
Comment (sowbiba) [inline tests-legacy/Unit/Core/Product/PriceFormatter.php:31]: I dont see where this test is moved

File: tests/Integration/Adapter/Module/PrestaTrust/PrestaTrustCheckerTest.php [hunk 5]
```diff
@@ -24,109 +24,101 @@
...
-            )),
-        );
-
-        $this->prestatrustApiResults = (object)array(
-            'hash_trusted' => true,
-            'property_trusted' => true,
-        );
+                ],
+            ], [
+                'path' => __DIR__ . '/../../../../Resources/modules/ganalytics/',
```
Comment (sowbiba) [inline tests/Integration/Adapter/Module/PrestaTrust/PrestaTrustCheckerTest.php:81]: maybe use `dirname(__DIR__, 3)`

File: tests/Integration/Core/Addon/Module/ModuleRepositoryTest.php [hunk 6]
```diff
@@ -24,149 +24,99 @@
...
                     )
                 ),
-                new FakeLogger(),
-                $this->translatorStub,
-                __DIR__.'/../../../../resources/modules/',
-            ))
-            ->setMethods(array('readCacheFile', 'generateCacheFile'))
+                $logger,
+                $translator,
+                __DIR__ . '/../../../../Resources/modules/',
```
Comment (sowbiba) [inline tests/Integration/Core/Addon/Module/ModuleRepositoryTest.php:95]: consider usage of dirname here too

File: tests/Integration/Classes/Assets/JavascriptManagerTest.php (hunk 1 above)
Comment (Progi1984) [reply tests/Integration/Classes/Assets/JavascriptManagerTest.php]: Thanks. I fix it.

File: tests/Integration/Classes/AccessTest.php [hunk 7]
```diff
@@ -85,90 +83,90 @@ public function testFindSlugByIdParentTab()
...
+    public function testSluggifyTab(): void
     {
-        $this->assertEquals('ROLE_MOD_TAB_ADMINCLASSNAMETEST_', Access::sluggifyTab(['class_name' => 'AdminClassNameTest']));
-        $this->assertEquals('ROLE_MOD_TAB_ADMINCLASSNAMETEST_READ', Access::sluggifyTab(['class_name' => 'AdminClassNameTest'], 'READ'));
-        $this->assertEquals('ROLE_MOD_TAB_ADMINCLASSNAMETEST_', Access::sluggifyTab(['class_name' => 'AdminClassNameTest', 'route_name' => null]));
-        $this->assertEquals('ROLE_MOD_TAB_ADMINCLASSNAMETEST_read', Access::sluggifyTab(['class_name' => 'AdminClassNameTest', 'route_name' => null], 'read'));
-
-        $this->assertEquals('ROLE_MOD_TAB_ADMINCLASSNAMETEST_', Access::sluggifyTab(['class_name' => 'AdminClassNameTest', 'route_name' => 'admin_route_name_test']));
-        $this->assertEquals('ROLE_MOD_TAB_ADMINCLASSNAMETEST_READ', Access::sluggifyTab(['class_name' => 'AdminClassNameTest', 'route_name' => 'admin_route_name_test'], 'READ'));
+        $this->assertEquals(
```
Comment (sowbiba) [inline tests/Integration/Classes/AccessTest.php:91]: Can't we have a dataProvider here ?

File: tests/Unit/Adapter/Module/AdminModuleDataProviderTest.php [hunk 8]
```diff
@@ -24,54 +24,62 @@
...
-    private $categoriesProviderS;
+    /**
+     * @var string
+     */
+    private const NOTICE = '[AdminModuleDataProvider] ';
+
+    /**
+     * @var AddonsInterface
+     */
+    private $addonsDataProvider;
```
Comment (sowbiba) [inline tests/Unit/Adapter/Module/AdminModuleDataProviderTest.php:51]: Please add blank line between attributes

**Comment by sowbiba** [comment]
Few minor feedbacks. Good job ;)

File: tests-legacy/Unit/Core/Product/PriceFormatter.php (hunk 4 above)
Comment (Progi1984) [reply tests-legacy/Unit/Core/Product/PriceFormatter.php:31]: THis test is not a test. I transform it in mock classes.

File: tests/Integration/Adapter/Module/PrestaTrust/PrestaTrustCheckerTest.php (hunk 5 above)
Comment (Progi1984) [reply tests/Integration/Adapter/Module/PrestaTrust/PrestaTrustCheckerTest.php:81]: Fixed

File: tests/Integration/Classes/AccessTest.php (hunk 7 above)
Comment (Progi1984) [reply tests/Integration/Classes/AccessTest.php:91]: Done

File: tests/Integration/Adapter/Cart/CartPresenterTest.php (hunk 3 above)
Comment (Progi1984) [reply tests/Integration/Adapter/Cart/CartPresenterTest.php:170]: As it is for testing private method (and it's not a good practice) I prefer divide code for removing it in the future.

**Review by sowbiba — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks for review

**Pull request merged**